src/main/java/com/conveyal/datatools/manager/auth/Auth0Users.java

```diff
@@ -119,6 +133,54 @@ private static String doRequest(URI uri) {
         return result;
     }
 
+    /**
+     * Gets an Auth0 API access token for authenticating requests to the Auth0 Management API. This will either create
+     * a new token using the oauth token endpoint or grab a cached token that it has already created (if it has not
+     * expired). More information on setting this up is here: https://auth0.com/docs/api/management/v2/get-access-tokens-for-production
+     */
+    private static String getApiToken() {
+        long nowInMillis = new Date().getTime();
+        // If cached token has not expired, use it instead of requesting a new one.
+        if (cachedToken != null && cachedToken.getExpirationTime() > nowInMillis) {
+            long minutesToExpiration = (cachedToken.getExpirationTime() - nowInMillis) / 1000 / 60;
+            LOG.info("Using cached token (expires in {} minutes)", minutesToExpiration);
+            return cachedToken.access_token;
+        }
+        LOG.info("Getting new Auth0 API access token (cached token does not exist or has expired).");
+        // Create client and build URL.
+        HttpClient client = HttpClientBuilder.create().build();
+        URIBuilder builder = getURIBuilder();
+        try {
+            // First get base url for use in audience URL param. (Trailing slash required.)
+            final String audienceUrl = builder.setPath(API_PATH + "/").build().toString();
+            URI uri = builder.setPath("/oauth/token").build();
+            LOG.info(uri.toString());
+            LOG.info(audienceUrl);
+            HttpPost post = new HttpPost(uri);
+            post.setHeader("content-type", "application/x-www-form-urlencoded");
+            List<NameValuePair> urlParameters = new ArrayList<>();
+            urlParameters.add(new BasicNameValuePair("grant_type", "client_credentials"));
+            urlParameters.add(new BasicNameValuePair("client_id", AUTH0_API_CLIENT));
+            urlParameters.add(new BasicNameValuePair("client_secret", AUTH0_API_SECRET));
+            urlParameters.add(new BasicNameValuePair("audience", audienceUrl));
+            post.setEntity(new UrlEncodedFormEntity(urlParameters));
+            HttpResponse response = client.execute(post);
+            int code = response.getStatusLine().getStatusCode();
+            if (code >= 300) LOG.error("Could not get Auth0 API token");
```

Comment:
I feel like there are a lot of different things that could go wrong in this method that don't have helpful error messages. Like if Auth0 somehow changes their access token response body, this method may begin returning null and there'd be some cryptic stack trace. It would be nice to see each of these exceptions encapsulated within a try/catch block that has a helpful error message and then either immediately returns null or re-throws the exception to a higher level.